py-utils/src/utils/message_bus/message_broker_collection.py

```
@@ -102,6 +105,84 @@ def init_client(self, client_type: str, **client_conf: dict):
+            num_partitions=partitions) for each_message_type in message_type]
+        created_message_types = admin.create_topics(new_message_type)
+        self._task_status(created_message_types)
+
+        for each_message_type in message_type:
+            for list_retry in range(1, self._max_list_message_type_count+2):
+                if each_message_type not in \
+                    list(self._get_metadata(admin).keys()):
+                    if list_retry > self._max_list_message_type_count:
+                        raise MessageBusError(errno.EINVAL, "Maximum retries \
+                            exceeded for creating %s.", each_message_type)
+                    time.sleep(list_retry*1)
+                    continue
+                else:
+                    break
+
+    def deregister_message_type(self, admin_id: str, message_type: list):
+        """ Deletes a message type """
+        admin = self._clients['admin'][admin_id]
+        deleted_message_types = admin.delete_topics(message_type)
+        self._task_status(deleted_message_types)
+
+        for each_message_type in message_type:
+            for list_retry in range(1, self._max_list_message_type_count+2):
+                if each_message_type in list(self._get_metadata(admin).keys()):
+                    if list_retry > self._max_list_message_type_count:
+                        raise MessageBusError(errno.EINVAL, "Maximum retries \
+                            exceeded for deleting %s.", each_message_type)
+                    time.sleep(list_retry*1)
+                    continue
+                else:
+                    break
+
+    def increase_parallelism(self, admin_id: str, message_type: list, \
+        partitions: int):
+        """ Increases the partitions for message type """
+        admin = self._clients['admin'][admin_id]
+        new_partition = [NewPartitions(each_message_type, \
+            new_total_count=partitions) for each_message_type in message_type]
```

What if the new count is less than the old count ? Should we report an error ?

---

yes, Kafka throws an error if the number of partitions is less than the existing number

---

We need to catch and throw a Message Bus Error. Can you confirm if this is the case ?

---

This exception is caught in **def _task_status(self, tasks :dict):** method